src/modules/hyprland/workspaces.cpp

```diff
@@ -207,8 +207,17 @@ void Workspaces::remove_workspace(std::string name) {
 }
 
 void Workspaces::fill_persistent_workspaces() {
-  if (config_["persistent_workspaces"].isObject() && !all_outputs()) {
-    const Json::Value persistent_workspaces = config_["persistent_workspaces"];
+  if (config_["persistent_workspaces"].isObject()) {
+    spdlog::warn(
+        "persistent_workspaces is deprecated. Please change config to use persistent-workspaces.");
+  }
+
+  if ((config_["persistent-workspaces"].isObject() ||
+       config_["persistent_workspaces"].isObject()) &&
+      !all_outputs()) {
```

Comment:
I wonder why I added `!all_outputs()` to here. I copied this from the `wlr/workspaces` module but I don't remember why I did that. Does it make sense to only allow persistent-workspaces when all_outputs is false?

Reply:
I personally think we should remove it, if you're okay with it. It feels counterintuitive... I am pinning a workspace to be displayed and if I want everything on all monitors I dont get to pin them?

Reply:
The only non-intuitive thing is that when it's removed the persistent workspace shows on the output you have pinned it to, but not the other outputs that all_outputs implies (until it isn't empty). But with the setting, you lose the persistent workspace and all outputs match in appearance.

Reply:
@zjeffer  Looks like you removed it already in that other PR when I rebased. :P

Reply:
>  Looks like you removed it already in that other PR when I rebased. :P

Lol I don't even remember doing that haha

Reply:
IIRC, I think the reason it's like that in `wlr/workspaces` was because some kind of limitation when persistent-workspaces was implemented in that module. 

@MonstrousOgre Do you remember why you did it this way in `wlr/workspaces`?

Reply:
What I saw when testing locally, was that you'll see the persistent workspace in the output it's mapped to but not the other outputs, as you might expect from enabling all_outputs. So, the logic doesn't account for empty persistent workspaces to be shown on other outputs. Might just be confusing until reworked a bit?

Reply:
> IIRC, I think the reason it's like that in `wlr/workspaces` was because some kind of limitation when persistent-workspaces was implemented in that module.
> 
> @MonstrousOgre Do you remember why you did it this way in `wlr/workspaces`?

So when all-outputs is true, the number of `WorkspaceManager` instances for each bar is equal to the number of monitors. In my case, this would be 2 for every bar. But since there are two bars (one for each monitor), there are totally 4 instances of `WorkspaceManager` per application.

So if I had `"5": ["eDP-1", "HDMI-A-1"],` in my config, workspace 5 would show up twice on every monitor, or 4 times totally. I figured I'd just prevent the all-outputs and persistent workspaces being used together to avoid this issue with duplicate workspaces